app/models/user.rb

```diff
@@ -1,7 +1,10 @@
 class User < ApplicationRecord
   has_secure_password
   validates :email, uniqueness: true, presence: true
-  validates_presence_of :password, require: true, :on => :create
+  validates_presence_of :password, require: true, :on => :create, :on => :update_password
+
+  validates_presence_of :password_confirmation, require: true, :on => :create, :on => :update_password
```

Comment:
Something we may have to look into is validating the confirmation against the password.

validates :password, confirmation: true

Reply:
@dhalverson the confirmation is built in